mantidproject/mantid
# Fix WorkspaceCreation::create(const P &parent) for parent workspaces with varying bins

`WorkspaceCreation::create(const P &parent)` always copied the X data of the first histogram in the parent workspace to the created one. This would lead to incorrect binning in the created workspace if the parent had varying bins.

**To test:**

The symptoms can be seen e.g. when using the [MonteCarloAbsorption](http://docs.mantidproject.org/nightly/algorithms/MonteCarloAbsorption-v1.html) algorithm. In the following script, change the `buildDir` variable to point to Mantid's build dir and run.
```python
buildDir = <path-to-Mantid-build-directory>
ws = Load(Filename=buildDir + 'ExternalData/Testing/Data/UnitTest/ILL/IN4/084446.nxs')
ws = ConvertUnits(ws, Target='Wavelength', EMode='Direct')
geometry = {
    'Shape': 'Cylinder',
    'Height': 10.0,
    'Radius': 0.5,
    'Center': [0.0, 0.0, 0.0]
}
material = {
    'ChemicalFormula': 'O2',
    'SampleMassDensity': 1
}
SetSample(ws, Geometry=geometry, Material=material)
corrections = MonteCarloAbsorption(ws, 
    NumberOfWavelengthPoints=2, 
    EventsPerPoint=1)
```
After the unit conversion `ws` has varying bins for workspace indices > 300. Binning should be the same for `corrections` after this PR has been applied.

Fixes #19242.

*Release notes were updated accordingly.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
